### ISSUE TITLE
What is the continent of Argentina, Brazil and Chile?

### ISSUE THREAD

**Issue by Ezibenroc** [ProjetPP/PPP-QuestionParsing-Grammatical#103]
Produces:
```
{
    "list": [
        {
            "subject": {
                "value": "Argentina", 
                "type": "resource"
            }, 
            "object": {
                "type": "missing"
            }, 
            "type": "triple", 
            "predicate": {
                "value": "continent", 
                "type": "resource"
            }
        }, 
        {
            "subject": {
                "value": "Brazil Chile", 
                "type": "resource"
            }, 
            "object": {
                "type": "missing"
            }, 
            "type": "triple", 
            "predicate": {
                "value": "continent", 
                "type": "resource"
            }
        }
    ], 
    "type": "intersection"
}
```

Reason: we merge nodes that we should not merge in the preprocessing.
![tmp](https://cloud.githubusercontent.com/assets/8900834/6050398/7a9377b4-acbf-11e4-92bb-0eec4d711f5f.png)

Tree given by the Stanford parser (before preprocessing):
![tmp](https://cloud.githubusercontent.com/assets/8900834/6050454/ece18fe0-acbf-11e4-97b5-2f00adf3aca2.png)

It looks great. It still works if you add elements to the conjunction (try to add Bolivia and Peru for instance).

Solution: do not merge nodes with `conj_*` dependency during the NER merging.

**Comment by yhamoudi** [comment]
I removed the merging for conjonctions between a son and its father for the same reason (here, nobody is merged with Argentina).

However, I don't know if it's relevant here. We have to do the choice that enables us to cover the most possible questions (or find a way to make a distinction between useful and useless merging...)

**Comment by Ezibenroc** [comment]
Now we don't merge these nodes in the preprocessing: https://github.com/ProjetPP/PPP-QuestionParsing-Grammatical/commit/ba36c8467d48ece99f2ae313ff536971782b2c99

We still produce a wrong tree:
![tmp](https://cloud.githubusercontent.com/assets/8900834/6060730/980c3ab6-ad40-11e4-9e4c-c03144e54164.png)

The tree after the preprocessing is as follow:
![tmp](https://cloud.githubusercontent.com/assets/8900834/6060744/bceeb872-ad40-11e4-8363-9564bd58f8bd.png)

**Comment by Ezibenroc** [comment]
I think the problem come from the handling of the `conj` dependency: we only considered conjunctions of two elements...

**Comment by Ezibenroc** [comment]
@yhamoudi what do you think of this adaptation of the algorithm to handle `conj` dependencies?
![conj_or-crop](https://cloud.githubusercontent.com/assets/8900834/6061437/1cc9fe08-ad48-11e4-8f79-91df8c1e04aa.png)

**Comment by yhamoudi** [comment]
it seems to be the natural way to extend what we do actually, so why not (perhaps try it on some examples before, it gives what we want on `What is the continent of Argentina, Brazil and Chile?` ?)

Actually the re-balanced operation for conjunction is performed by [conjConnectorsUp](https://github.com/ProjetPP/PPP-QuestionParsing-Grammatical/blob/master/ppp_questionparsing_grammatical/dependencyAnalysis.py#L181) (the most ugly function ever...). We can try to adapt the code (not so easy), or rewrite it more properly (if you have an idea...)

**Comment by Ezibenroc** [comment]
> it gives what we want on `What is the continent of Argentina, Brazil and Chile?` ?

Yes. It gives a subtree with a node `and` and 3 children: `Argentina`, `Brazil` and `Chile`.
We would then have to adapt the function `Normalize` to handle an arbitrary number of children for the nodes `and` and `or`, but this is very natural.

> Actually the re-balanced operation for conjunction is performed by conjConnectorsUp (the most ugly function ever...). We can try to adapt the code (not so easy), or rewrite it more properly (if you have an idea...)

Yes it would be better to rewrite it. I will think about that.

**Comment by yhamoudi** [comment]
I realize that the article isn't true on the transformation. Here is what we really do in case of 2 elements (the generalisation is straightforward):
![image](https://cloud.githubusercontent.com/assets/6659728/6061909/ee06bd8c-ad4b-11e4-8b33-433892b1b111.png)
